kernel/poll.c

```diff
@@ -246,7 +246,7 @@ int k_poll(struct k_poll_event *events, int num_events, s32_t timeout)
 
 	_pend_current_thread(&wait_q, timeout);
 
-	int swap_rc = _Swap(key);
+	rc = _Swap(key);
```

Comment:
Why?  This surely doesn't affect the generated code at all (and if so only spuriously by e.g. changing register assignments), and honestly IMHO it was clearer with one specific/named register per use.   Mutable variables are a bad thing when not needed.

Reply:
I always thought this would require another register, isn't that a common practice not to introduce new variable if you can reuse them. Perhaps this depends on the compiler and if it can detect there is no overlapping of rc and swap_rc, but depending on the compiler to do the right thing sometimes can backfire.